core/src/main/scala/org/apache/spark/shuffle/ShuffleManager.scala

```diff
@@ -43,23 +44,15 @@ private[spark] trait ShuffleManager {
       context: TaskContext,
       metrics: ShuffleWriteMetricsReporter): ShuffleWriter[K, V]
 
-  /**
-   * Get a reader for a range of reduce partitions (startPartition to endPartition-1, inclusive).
-   * Called on executors by reduce tasks.
-   */
-  def getReader[K, C](
```

Comment:
can we keep this method as an overload? We can mark it as final and simply call the other `getReader` with `0, Int.Max` as the map index parameters.

Reply:
ok, sounds good!